### PR TITLE
Add verify-build step to make target

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
This new targets builds the project for a couple of chosen platforms,
which should hint us in the future if any dependency may not work for
one of those.
#### Which issue(s) this PR fixes:

Refers to https://github.com/kubernetes/release/issues/1862
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
`pull-release-verify` tells us that it works:
```
…
Building project for linux/386
Building project for linux/arm
Building project for linux/arm64
Building project for linux/ppc64le
Building project for linux/s390x
Building project for windows/amd64
go: downloading github.com/inconshreveable/mousetrap v1.0.0
go: downloading github.com/Azure/go-ansiterm v0.0.0-20170929234023-d6e3b3328b78
go: downloading github.com/StackExchange/wmi v0.0.0-20190523213315-cbe66965904d
go: downloading github.com/go-ole/go-ole v1.2.4
Building project for windows/386
Building project for freebsd/amd64
…
```

**Comment by xmudrii** [comment]
It doesn't take longer than pull-release-cluster-up, so I don't think it matters that it runs a little bit longer. :)

File: Makefile [hunk 1]
```diff
@@ -37,11 +37,14 @@ verify-published-rpms: ## Ensure rpms have been published
...
 # TODO: Uncomment verify-shellcheck once we finish shellchecking the repo.
 #       ref: https://github.com/kubernetes/release/issues/726
-verify: release-tools verify-boilerplate verify-dependencies verify-golangci-lint verify-go-mod #verify-shellcheck ## Runs verification scripts to ensure correct execution
+verify: release-tools verify-boilerplate verify-build verify-dependencies verify-golangci-lint verify-go-mod #verify-shellcheck ## Runs verification scripts to ensure correct execution
 
 verify-boilerplate: ## Runs the file header check
 	./hack/verify-boilerplate.sh
 
+verify-build: ## Builds the project for a chosen set of platforms
+	./hack/verify-build.sh
```
Comment (xmudrii) [inline Makefile:46]: I think this target should be added as a PHONY target on [L26](https://github.com/kubernetes/release/blob/bc9c5bf82886807b005a6d7a17cf97d725e63770/Makefile#L26).

Comment (saschagrunert) [reply Makefile:46]: Fixed :+1:

**Review by xmudrii — APPROVED** [review]
Looks good to me.

/lgtm

**Pull request merged**